### DISCUSSION TITLE
Error while running composer require uneca/dashboard-starter-kit

### DISCUSSION THREAD

**Discussion by workualemu** [tech-acs/chimera-docs#8] [Q&A] [answered]
While creating new dashboard project,  composer require uneca/dashboard-starter-kit is generating error in Windows environment.  Error message is not clear.

C:\workspace_eca\uganda-census-dashboard> composer require uneca/dashboard-starter-kit
Info from https://repo.packagist.org: #StandWithUkraine
Using version ^2.5 for uneca/dashboard-starter-kit
./composer.json has been updated
Running composer update uneca/dashboard-starter-kit
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - uneca/dashboard-starter-kit[2.5.0, ..., 2.5.20] require illuminate/contracts ^9.0 -> found illuminate/contracts[v9.0.0, ..., v9.52.4] but these were not loaded, likely because it conflicts with another require.
    - uneca/dashboard-starter-kit[2.5.21, ..., 2.7.1] require ext-redis * -> it is missing from your system. Install or enable PHP's redis extension.
    - Root composer.json requires uneca/dashboard-starter-kit ^2.5 -> satisfiable 
by uneca/dashboard-starter-kit[2.5.0, ..., 2.7.1].

To enable extensions, verify that they are enabled in your .ini files:
    - C:\php\php.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-redis` to tempYou can also try re-running composer require with an explicit version constraint, 
e.g. "composer require uneca/dashboard-starter-kit:*" to figure out if any version is installable, or "composer require uneca/dashboard-starter-kit:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by workualemu** [comment, marked as answer]
I had this issue due to redis extension.  Running the below command worked for me.

composer update uneca/dashboard-starter-kit --ignore-platform-req=ext-redis

**Reply by amestsantim** [reply]
In the situation where you are not able to install the redis extension, you can ignore the requirement as you have and then use the predis package.

https://laravel.com/docs/10.x/redis#predis

**Reply by baakal** [reply]
Additional note: the issue occurs on `php artisan chimera:install` as well, to overcome this particular problem, since this command does a composer update internally, to resolve the problem we used the following workaround: setting the environment variables before running the command `COMPOSER_IGNORE_PLATFORM_REQ=ext-redis`,
